### PR TITLE
Add tier 1 CRD namespace tests to test_namespace

### PR DESCRIPTION
**Pull request by romayalon**
Signed-off-by: Romy <romy2232@gmail.com>

1. Added ns_store factory.
2. Added NamespaceStore template and updated the bucketclass template.
2. Added namespace policy for bucketclass creation (used for the creation of namespace buckets)
3. Updated tier1 tests in test_namespace.py
4. Fixed a bug when creating a secret for s3_client of type RGW - noobaa expects for "AWS_ACCESS_KEY_ID"
and "AWS_SECRET_ACCESS_KEY" and not RGW prefix.

### CODE REVIEW

File: tests/manage/mcg/test_namespace.py [hunk 1]
```diff
@@ -30,83 +31,185 @@ class TestNamespace(MCGTest):
...
+        }
+        if namespace_policy_type == constants.NAMESPACE_POLICY_TYPE_SINGLE:
+            bucketclass_dict["namespace_policy_dict"]["resource"] = ns_store_name
+        elif namespace_policy_type == constants.NAMESPACE_POLICY_TYPE_MULTI:
+            bucketclass_dict["namespace_policy_dict"]["read_resources"] = [
+                ns_store_name
+            ]
+            bucketclass_dict["namespace_policy_dict"]["write_resource"] = ns_store_name
+
+        bucketclass = bucket_class_factory(bucketclass_dict)
```
Comment (MeridianExplorer) [inline tests/manage/mcg/test_namespace.py:101]: As part of #3437, all the bucket setup (including bucketclasses, backingstores and ULS) is happening behind the scenes as part of `bucket_factory`.
Please migrate the NSS creation/deletion/manipulation logic to work with this flow

**Review by MeridianExplorer — COMMENTED** [review]
Phenomenal job 👌🏻 ✨

File: tests/manage/mcg/test_namespace.py [hunk 2]
```diff
@@ -30,83 +31,185 @@ class TestNamespace(MCGTest):
...
             amount=1,
-            interface="mcg-namespace",
-            write_ns_resource=ns_resource_name,
-            read_ns_resources=[ns_resource_name],
+            interface="OC",
+            bucketclass=bucketclass.name,
         )
 
-    @pytest.mark.polarion_id("OCS-2407")
+    @skipif_ocs_version("<4.7")
```
[the remark this answers is not in the document]
Comment (MeridianExplorer) [reply tests/manage/mcg/test_namespace.py:108]: I discussed the matter with Elad on the phone - we have to continue supporting 4.6 for the foreseeable future; Thus, we have to duplicate all tests and methods, and split them to `OC` (CRD) and `RPC`.

[the remark this answers is not in the document]
Comment (romayalon) [reply tests/manage/mcg/test_namespace.py:107]: sure, duplicated the tests.

File: ocs_ci/ocs/resources/mcg.py [hunk 3]
```diff
@@ -593,7 +593,58 @@ def create_namespace_resource(
...
+            str: The name of the created target_bucket_name (cloud uls)
+        """
+        # Create the actual target bucket on AWS
+        uls_dict = cloud_uls_factory({platform: [(1, region)]})
+        target_bucket_name = list(uls_dict[platform])[0]
+
+        ns_data = templating.load_yaml(constants.MCG_NAMESPACESTORE_YAML)
+        ns_data["metadata"]["name"] = ns_store_name
+        ns_data["metadata"]["namespace"] = config.ENV_DATA["cluster_namespace"]
+        if platform == constants.AWS_PLATFORM:
```
[the remark this answers is not in the document]
Comment (romayalon) [reply ocs_ci/ocs/resources/mcg.py]: sure, fixed.

File: ocs_ci/ocs/resources/namespacestore.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        created_ns_stores.append(
+            NamespaceStore(
+                name=rand_ns_store,
+                method="oc",
+                mcg_obj=mcg_obj,
+            )
+        )
+        return target_bucket_name, rand_ns_store
+
+    def ns_stores_cleanup():
```
[the remark this answers is not in the document]
Comment (romayalon) [reply ocs_ci/ocs/resources/namespacestore.py]: fixed.

File: ocs_ci/ocs/resources/namespacestore.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        for ns_store in created_ns_stores:
```
[the remark this answers is not in the document]
Comment (romayalon) [reply ocs_ci/ocs/resources/namespacestore.py]: fixed.

File: ocs_ci/ocs/resources/namespacestore.py [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+            try:
+                ns_store.delete()
```
[the remark this answers is not in the document]
Comment (romayalon) [reply ocs_ci/ocs/resources/namespacestore.py]: fixed.

File: tests/manage/mcg/test_namespace.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (romayalon) [reply tests/manage/mcg/test_namespace.py:107]: fixed.

File: ocs_ci/ocs/resources/mcg.py [hunk 7]
```diff
@@ -668,9 +721,42 @@ def oc_create_bucketclass(self, name, backingstores, placement):
...
         bc_data["metadata"]["namespace"] = self.namespace
-        tiers = bc_data["spec"]["placementPolicy"]["tiers"][0]
-        tiers["backingStores"] = backingstores
-        tiers["placement"] = placement
+        if (backingstores is not None) and (placement is not None):
+            tiers = bc_data["spec"]["placementPolicy"]["tiers"][0]
+            tiers["backingStores"] = backingstores
+            tiers["placement"] = placement
+        else:
+            del bc_data["spec"]["placementPolicy"]
```
[the remark this answers is not in the document]
Comment (romayalon) [reply ocs_ci/ocs/resources/mcg.py]: ok no problem, fixed.

File: ocs_ci/ocs/resources/mcg.py [hunk 8]
```diff
@@ -593,7 +593,60 @@ def create_namespace_resource(
...
+            platform (str): The platform resource name
+
+        Returns:
+            str: The name of the created target_bucket_name (cloud uls)
+        """
+        # Create the actual target bucket on AWS
+        uls_dict = cloud_uls_factory({platform: [(1, region)]})
+        target_bucket_name = list(uls_dict[platform])[0]
+
+        ns_data = templating.load_yaml(constants.MCG_NAMESPACESTORE_YAML)
```
Comment (MeridianExplorer) [inline ocs_ci/ocs/resources/mcg.py:618]: I think it'd be best to define `NS` as `NAMESPACE` and `NSS` as `NAMESPACESTORE`.
Can you please refactor accordingly?

File: tests/conftest.py [hunk 9]
```diff
@@ -2910,6 +2915,22 @@ def ns_resources_cleanup():
     return _create_ns_resources
 
 
+@pytest.fixture()
```
Comment (MeridianExplorer) [inline tests/conftest.py:2918]: Please add a `session`-scoped factory
You can base it on `bucket_factory()` and `bucket_factory_session()` (make sure to also use the session-scoped fixtures in the parameters)

File: ocs_ci/ocs/resources/mcg.py (hunk 8 above)
Comment (romayalon) [reply ocs_ci/ocs/resources/mcg.py]: sure, fixed

File: tests/conftest.py (hunk 9 above)
Comment (romayalon) [reply tests/conftest.py:2892]: added according to backingstore_factory

File: tests/manage/mcg/test_namespace.py [hunk 10]
```diff
@@ -30,83 +31,185 @@ class TestNamespace(MCGTest):
...
+            bucketclass_dict["namespace_policy_dict"]["write_resource"] = ns_store_name
+
+        bucketclass = bucket_class_factory(bucketclass_dict)
         bucket_factory(
             amount=1,
-            interface="mcg-namespace",
-            write_ns_resource=ns_resource_name,
-            read_ns_resources=[ns_resource_name],
+            interface="OC",
+            bucketclass=bucketclass.name,
```
[the remark this answers is not in the document]
Comment (romayalon) [reply tests/manage/mcg/test_namespace.py]: yes, fixed to dictionaries

**Comment by MeridianExplorer** [comment]
Work continues in #3620

**Pull request closed without merging**